### PR TITLE
rkt: bump rkt api service version

### PR DESCRIPTION
**Pull request by yifan-gu**
Depends on #18643 The last three commits are new

### CODE REVIEW

**Comment by yifan-gu** [comment]
cc @yujuhong @Random-Liu @dgonyeo @jonboulle

File: pkg/kubelet/rkt/rkt.go [hunk 1]
```diff
@@ -400,24 +403,37 @@ func setApp(app *appctypes.App, c *api.Container, opts *kubecontainer.RunContain
...
+			},
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("couldn't list images: %v", err)
+	}
+	return listResp.Images, nil
+}
+
 // getImageManifest invokes 'rkt image cat-manifest' to retrive the image manifest
```
Comment (jonboulle) [inline pkg/kubelet/rkt/rkt.go]: update

File: pkg/kubelet/rkt/rkt.go [hunk 2]
```diff
@@ -1459,35 +1417,24 @@ func (r *Runtime) getPodInfo(uuid string) (*podInfo, error) {
...
-// getImageByName tries to find the image info with the given image name.
-// TODO(yifan): Replace with 'rkt image cat-manifest'.
-// imageName should be in the form of 'example.com/app:latest', which should matches
-// the result of 'rkt image list'. If the version is empty, then 'latest' is assumed.
-func (r *Runtime) getImageByName(imageName string) (*kubecontainer.Image, error) {
-	// TODO(yifan): Print hash in 'rkt image cat-manifest'?
-	images, err := r.ListImages()
+// buildImageName constructs the image name for kubecontainer.Image.
+func buildImageName(img *rktapi.Image) string {
+	return fmt.Sprintf("%s:%s", img.Name, img.Version)
```
Comment (jonboulle) [inline pkg/kubelet/rkt/rkt.go]: Version is optional, isn't it?

File: pkg/kubelet/rkt/rkt.go [hunk 3]
```diff
@@ -1510,27 +1457,236 @@ func (r *Runtime) ListImages() ([]kubecontainer.Image, error) {
...
+	return podStatus, nil
 }
 
-func (r *Runtime) ConvertPodStatusToAPIPodStatus(_ *api.Pod, _ *kubecontainer.PodStatus) (*api.PodStatus, error) {
-	return nil, fmt.Errorf("Not implemented yet")
+type sortByRestartCount []*kubecontainer.ContainerStatus
+
+func (s sortByRestartCount) Len() int           { return len(s) }
+func (s sortByRestartCount) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+func (s sortByRestartCount) Less(i, j int) bool { return s[i].RestartCount > s[j].RestartCount }
```
Comment (jonboulle) [inline pkg/kubelet/rkt/rkt.go]: Hm, this is confusing - maybe we should tweak the naming of this, or have it sort actually less-than, then use sort.Reverse instead

Comment (yifan-gu) [inline pkg/kubelet/rkt/rkt.go]: Ok, let me do Reverse

File: pkg/kubelet/rkt/rkt.go (hunk 2 above)
Comment (yifan-gu) [inline pkg/kubelet/rkt/rkt.go]: @jonboulle The result returned by api service will always contain the version. "latest" by default.

**Comment by yifan-gu** [comment]
@jonboulle Ready for another review :)

**Comment by yifan-gu** [comment]
Add one more commits to delete dead codes as we don't have to save/read things to/from the service files anymore.

**Comment by yujuhong** [comment]
LGTM with minor nit/question. Travis failed because godeps is broken by docker pkg move #18833

**Comment by yifan-gu** [comment]
> In what cases an image name will be mapped to more than one images? getImageManifest seems to assume one image?

Good question. currently rkt allows multiple images with the same name (but different hash). So it can be multiple images. 
Let me see if I can make sure it returns the latest imported one.

**Comment by yifan-gu** [comment]
@yujuhong Updated. 
I am not sure we should prevent multiple same image names, but anyway I noted done it.

**Comment by yujuhong** [comment]
LGTM.

**Comment by yifan-gu** [comment]
@k8s-bot test this please

**Comment by yifan-gu** [comment]
Not sure why travis doesn't build, rebased the master.

**Comment by yujuhong** [comment]
Sorry for removing the cla label. The response time of github label is usually too long when using the keyboard shortcuts. It never seems to do what I want unless I pause on every key stroke :-(

**Comment by yifan-gu** [comment]
@yujuhong No worries :)

**Comment by yifan-gu** [comment]
@k8s-bot test this please

**Comment by yifan-gu** [comment]
@k8s-bot test this please

**Comment by smarterclayton** [comment]
I think this was the last change to bump Godeps, but now changes that invoke hack/verify-godeps.sh are failing with:

```
$ hack/verify-godeps.sh master
Starting to download all kubernetes godeps. This takes a while
godep: restore, during download dep "github.com/camlistore/camlistore/pkg/errorutil": exit status 1
```

As in: https://travis-ci.org/kubernetes/kubernetes/builds/98713963

Did something change in camlistore?

**Comment by jonboulle** [comment]
Yes, looks like they [moved](https://github.com/camlistore/camlistore/commit/40be4d8da29a164513347d54334a3aed849585db) that dependency into a [different repo](https://github.com/camlistore/go4/tree/master/errorutil) - and godep can't (currently) handle a package that's disappeared from the HEAD of a repo (could have sworn it used to..)

**Pull request merged**